### ISSUE TITLE
Mermaid usage

### ISSUE THREAD

**Issue by slavamirniy** [obsidian-html/obsidian-html#18]
Mermaid saves as code, no image.

**Comment by slavamirniy** [comment]
For example, now i just add to files with mermaid this code
```
<script>
    var codes = document.querySelectorAll(".codehilite")
    var i = 0;
    for(j = 0; j < codes.length; j++){
    	codeElement = codes[j]
    	console.log(codeElement)
        if(codeElement.textContent.match("graph") == null)
            continue;
        i++
        codeElement.innerHTML = codeElement.textContent
        codeElement.className = "mermaid"
    }
</script>  
<script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>```

**Comment by dwrolvink** [comment]
Hi @slavamirniy thank you for your feature request!

I've added support for mermaid in version 0.0.7: https://pypi.org/project/obsidianhtml/0.0.7/ (https://github.com/obsidian-html/obsidian-html/commit/6fb743445d7c21bc68dd514e089945445ac02a01)
You can see a very minimal proof of concept here: https://obsidian-html.github.io/#mermaid

I don't really use mermaid yet, so I don't have any good test cases, let me know if this change works for you.

This adds the requirement of md-mermaid, which will be installed automatically if you follow the pip install route with the newest version. If you use the git repo code directly don't forget to run `pip install md-mermaid`. 

Two new files are added to the root of the html output folder: mermaid.css & mermain.min.js _(the js file is the exact same file as the one you used, I just like to minimize external dependencies)_.

**Comment by dwrolvink** [comment]
Just noticed there is a bug when the code is located on a tab that is opened after the page is done loading, will see how to fix that.

**Comment by dwrolvink** [comment]
Okay, fixed in 0.0.8

Compare:
https://obsidian-html.github.io/?path=%2FResources%2FExample%2520site%2FesMermaid.html/ (dynamically loaded)
https://obsidian-html.github.io/Resources/Example%20site/esMermaid.html (loaded normally).

mermaid.init() was the solution here.

**Comment by slavamirniy** [comment]
Thanks you!

**Comment by slavamirniy** [comment]
> Okay, fixed in 0.0.8
> 
> Compare: https://obsidian-html.github.io/?path=%2FResources%2FExample%2520site%2FesMermaid.html/ (dynamically loaded) https://obsidian-html.github.io/Resources/Example%20site/esMermaid.html (loaded normally).
> 
> mermaid.init() was the solution here.

I tested it. Now it deletes all Russian symbols in files.

**Comment by dwrolvink** [comment]
Compare
``` python
html_body = markdown.markdown(md.page, extensions=['extra', 'codehilite', 'toc', 'md_mermaid'], extension_configs=extension_configs)
```
vs 
```
html_body = markdown.markdown(md.page, extensions=['extra', 'codehilite', 'toc'], extension_configs=extension_configs)
```

The first case retains russian characters, the second case removes them. So there is something wrong with the mermaid plugin.

For now a workaround/fix is making this change here: https://github.com/obsidian-html/obsidian-html/blob/af75811442d502e3fd32c7713591200fc3ba4ebf/obsidianhtml/__init__.py#L218

Though that will break mermaid support.

Will investigate if there is a solution. Perhaps use a different plugin/mermaid parse method.
